osu.Android/OsuGameActivity.cs

```diff
@@ -23,8 +30,37 @@ protected override void OnCreate(Bundle savedInstanceState)
 
             base.OnCreate(savedInstanceState);
 
+            OnNewIntent(Intent);
+
             Window.AddFlags(WindowManagerFlags.Fullscreen);
             Window.AddFlags(WindowManagerFlags.KeepScreenOn);
         }
+
+        protected override void OnNewIntent(Intent intent)
+        {
+            if (intent.Action == Intent.ActionDefault)
+            {
+                if (intent.Scheme == ContentResolver.SchemeContent)
+                        handleImportFromUri(intent.Data);
+            }
+
+            if (intent.Action == Intent.ActionSend)
+            {
+                var content = intent.ClipData.GetItemAt(0);
+                handleImportFromUri(content.Uri);
+            }
+        }
+
+        private void handleImportFromUri(Uri uri)
+        {
+            var cursor = ContentResolver.Query(uri, null, null, null);
```

Comment:
[To quote docs](https://developer.android.com/reference/android/content/ContentResolver#query(android.net.Uri,%20java.lang.String[],%20android.os.Bundle,%20android.os.CancellationSignal)), the second argument is:

> A list of which columns to return. Passing null will return all columns, which is inefficient.

Maybe this should just fetch the filename only?